### ISSUE TITLE
clang-15 frontend crash

### ISSUE THREAD

**Issue by hkaelber** [llvm/llvm-project#68402]
We experience following repeating crash with clang-15:
```
[build] 1.	/tdtp-src/tests/udmd/unit/test_Transfer.cpp:179:74: current parser token ')'
[build] 2.	/tdtp-src/tests/udmd/unit/test_Transfer.cpp:20:1: parsing namespace 'udmd'
[build] 3.	/tdtp-src/tests/udmd/unit/test_Transfer.cpp:71:1: parsing function body 'udmd::tests::CATCH2_INTERNAL_TEST_1'
[build] 4.	/tdtp-src/tests/udmd/unit/test_Transfer.cpp:71:1: in compound statement ('{}')
[build] 5.	/tdtp-src/tests/udmd/unit/test_Transfer.cpp:171:5: in compound statement ('{}')
[build] 6.	/tdtp-src/tests/udmd/unit/test_Transfer.cpp:178:9: in compound statement ('{}')
[build] clang-15: error: clang frontend command failed with exit code 139 (use -v to see invocation)
[build] Alpine clang version 15.0.7
[build] Target: aarch64-poky-linux
```
Any hints how to work around this?

[test_Transfer-a442e1.txt](https://github.com/llvm/llvm-project/files/12829951/test_Transfer-a442e1.txt)
[test_Transfer-a442e1.zip](https://github.com/llvm/llvm-project/files/12829958/test_Transfer-a442e1.zip)

**Comment by DimitryAndric** [comment]
No idea, can you provide a reproduction test case? What happens if you use clang 16, 17, or recent main?

**Comment by EugeneZelenko** [comment]
Could you please try 17 or `main` branch?

**Comment by hkaelber** [comment]
Thx for your feedback, I'll try, but it's not straight forward as clang is part of our cross-compilation env and not simply exchangable. So far did not succeed.

**Comment by DimitryAndric** [comment]
This is a duplicate of #57170 ([clang] assertion `isDependentContext() && "cannot iterate dependent diagnostics of non-dependent context"' failed), which was fixed in ea4fd668c2cd88d13b36a5d64e3dedb1106340bc ([clang] fix crash on generic lambda with lambda in decltype), so for clang >= 16.0.

**Comment by hkaelber** [comment]
Thx for your analysis, @DimitryAndric. Can confirm that at least clangd-16 does not crash anymore when upgraded by the vscode plugin "clangd" (it crashed with clangd-15). This helps us at least to track down the issue and possibly work around.
